src/workerd/api/global-scope.c++

```diff
@@ -476,10 +480,15 @@ kj::Promise<WorkerInterface::AlarmResult> ServiceWorkerGlobalScope::runAlarm(
         // We don't usually log these messages, but it's useful to know the real reason we failed
         // to correctly investigate stuck alarms.
         LOG_NOSENTRY(ERROR, "output lock broke during alarm execution without an interesting error description", actorId, e);
+        if (e.getType() == kj::Exception::Type::OVERLOADED) {
+          // The handler failed because the user overloaded the object. It's their fault, we'll not
```

Comment:
Overloaded errors are not currently always the user's fault, so I don't think we can use this broad of a check without tightening up where OVERLOADED errors can happen.

To be specific, one case that I have in mind is that if storage operations slow way down (e.g. if the underlying storage layer is unhealthy), then this particular OVERLOADED error gets thrown and the user really has no ability to do anything about it: https://github.com/cloudflare/workerd/blob/main/src/workerd/io/worker.c%2B%2B#L3016-L3018

Reply:
Given that we saw exceededCpu errors, is there a way to check for that specific kind of overload?

Reply:
> Overloaded errors are not currently always the user's fault, so I don't think we can use this broad of a check without tightening up where OVERLOADED errors can happen.
> 
> To be specific, one case that I have in mind is that if storage operations slow way down (e.g. if the underlying storage layer is unhealthy), then this particular OVERLOADED error gets thrown and the user really has no ability to do anything about it: [`main`/src/workerd/io/worker.c++#L3016-L3018](https://github.com/cloudflare/workerd/blob/main/src/workerd/io/worker.c%2B%2B?rgh-link-date=2024-07-23T19%3A07%3A16Z#L3016-L3018)

For this specific error case, we can match either the description (which I don't think is a good practice, but would catch this specific case for now) or check for the presence of `broken.exceededMemory`...